File: 1167-minimum-cost-to-connect-sticks/1167-minimum-cost-to-connect-sticks.cpp
class Solution {
public:
    int connectSticks(vector<int>& sticks) {
        
        priority_queue<int, vector<int>, greater<int>>pq;
        
        for(int stick : sticks)
            pq.push(stick);
        int res = 0;
        while(pq.size()>1)
        {
            int x = pq.top();
            pq.pop();
            int y = pq.top();
            pq.pop();
            res+= x+y;
            pq.push(x+y);
        }
        return res;
    }
};

File: 1167-minimum-cost-to-connect-sticks/1167-minimum-cost-to-connect-sticks.py
from queue import PriorityQueue
class Solution:
    def connectSticks(self, sticks: List[int]) -> int:
        
        pq = PriorityQueue()
              
        for stick in sticks:
            pq.put(stick)   
        res = 0
        while(pq.qsize()> 1):
            
            stick1 = pq.get()
            stick2 = pq.get()
            res+= stick1+stick2
            pq.put(stick1 + stick2)
            
        return res
            
            
        